src/Features/Core/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -54,7 +54,12 @@ public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this Diag
                         category |= DiagnosticAnalyzerCategory.ProjectAnalysis;
                     }
 
-                    if (!cantSupportSemanticSpanAnalysis)
+                    // Current DiagnosticAnalyzer API has no restriction on any analyzer action reporting diagnostics anywhere within a compilation.
+                    // Hence, we must conservatively assume they can't support incremental span based analysis across compilations.
+                    // GitHub issue https://github.com/dotnet/roslyn/issues/1565 tracks adding support for marking an analyzer incremental.
+                    // Until then, we just assume that all public/non-built in analyzers can never support incremental analysis.
+
+                    if ((analyzer is IBuiltInAnalyzer) && !cantSupportSemanticSpanAnalysis)
```

Comment:
even when it doesn't have compilation (end) action? I think I am not understanding correctly.